tests/integration_tests/modules/test_ssh_keys_provided.py

```diff
@@ -107,6 +108,13 @@ def test_ssh_rsa_keys_provided(self, class_client):
             "4DOkqNiUGl80Zp1RgZNohHUXlJMtAbrIlAVEk+mTmg7vjfyp2un"
             "RQvLZpMRdywBm") in out
 
+    def test_ssh_rsa_certificate_provided(self, class_client):
+        """Test rsa certificate was imported."""
+        out = class_client.read_from_file("/etc/ssh/ssh_host_rsa_key-cert.pub")
+        assert (
+            "AAAAHHNzaC1yc2EtY2VydC12MDFAb3BlbnNzaC5jb20AAAAgMpg"
+            "BP4Phn3L8I7Vqh7lmHKcOfIokEvSEbHDw83Y3JloAAAAD") in out
+
```

Comment:
```suggestion

    def test_ssh_certifcate_updated_sshd_config(self, class_client):
        """Test ssh certificate was added to /etc/ssh/sshd_config."""
        out = class_client.read_from_file("/etc/ssh/sshd_config").strip()
        assert "HostCertificate /etc/ssh/ssh_host_rsa_key-cert.pub" in out

```

Reply:
Thanks for writing the tests :) I had to make some changes and then made two more:

 * The tests, as they were written, will read the real `/etc/ssh/sshd_config`. This causes problems since this means the expected value isn't written back (a superset is written back). Since multiple files are actually read while `handle()`-ing, we have to only mock the `open()`ing of that file so I wrote a hack. I don't know if that's too hack-is.
 * I refactored out the `GENERATE_KEY_NAMES` throughout the test module and added a version where DSA was filtered out, as required.